docs/src/components/productX/XComponentsSwitcher.tsx

```diff
@@ -0,0 +1,177 @@
+        gap: 2.5,
+      }}
+    >
+      <div>{icon}</div>
+      <div>
+        <Typography
+          component="span"
+          color="text.primary"
+          variant="body2"
+          fontWeight="bold"
+          display="block"
+        >
+          {name}
+        </Typography>
+        {description && (
+          <Typography
+            component="span"
+            color="text.secondary"
+            variant="body2"
+            fontWeight="regular"
+            display="block"
+            gutterBottom
+          >
+            {description}
+          </Typography>
+        )}
+        <Link
+          href={href}
+          color="primary"
+          variant="body2"
+          sx={{
+            display: 'inline-flex',
+            alignItems: 'center',
+            fontWeight: 'semiBold',
+            '& > svg': { transition: '0.2s' },
+            '&:hover > svg': { transform: 'translateX(2px)' },
+          }}
+          onClick={(event: React.MouseEvent<HTMLAnchorElement>) => {
+            event.stopPropagation();
+          }}
+        >
+          <span>View the docs</span>{' '}
+          <Box component="span" sx={visuallyHidden}>
+            {label}
+          </Box>
+          <KeyboardArrowRightRounded fontSize="small" sx={{ mt: '1px', ml: '2px' }} />
+        </Link>
+      </div>
+    </Box>
+  );
+}
+
+export default function XComponentsSwitcher(props: {
+  inView?: boolean;
+  componentIndex: number;
+  setComponentIndex: React.Dispatch<React.SetStateAction<number>>;
+}) {
+  const { inView = false, componentIndex, setComponentIndex } = props;
+  const productElements = [
+    <ComponentItem
+      name="Data Grid"
+      label="by going to the Core components page"
+      description="Feature-rich and fast table extension."
+      icon={<PivotTableChartRoundedIcon />}
+      href={ROUTES.dataGridOverview}
+    />,
+    <ComponentItem
+      name="Date and Time Pickers"
+      description="Let users pick a date and time, or both together."
+      label="by going to the Advanced components page"
+      icon={<CalendarMonthRoundedIcon />}
+      href={ROUTES.datePickersOverview}
+    />,
+    <ComponentItem
+      name="Charts"
+      description="Features bar, lines, pie, scatter, and more types of graphs."
+      label="by going to the templates page"
+      icon={<BarChartRoundedIcon />}
+      href={ROUTES.chartsOverview}
+    />,
+    <ComponentItem
+      name="Tree View"
+      description="Display hierarchical data, such as a file system navigator."
+      label="by going to the design-kits page"
+      icon={<AccountTreeRounded />}
+      href={ROUTES.treeViewOverview}
+    />,
+  ];
+  return (
+    <React.Fragment>
+      <Box
+        sx={{
+          display: { md: 'none' },
+          maxWidth: 'calc(100vw - 40px)',
+          minHeight: { xs: 200, sm: 166 },
+          '& > div': { pr: '32%' },
+        }}
+      >
+        {inView && (
+          <SwipeableViews
```

Comment:
This doesn't seem to be working:
<img width="591" src="https://github.com/mui/material-ui/assets/13808724/92e4c67f-c7e4-41a3-9901-53b4078e7fc8">

Reply:
I think it's solved now — thanks for spotting it! 🤙